xmake/rules/c++/modules/modules_support/clang/builder.lua

```diff
@@ -212,22 +212,21 @@ function make_module_buildjobs(target, batchjobs, job_name, deps, opt)
 
             local build, dependinfo
             local dependfile = target:dependfile(bmifile or opt.objectfile)
-            if provide or compiler_support.has_module_extension(opt.cppfile) then
-                build, dependinfo = should_build(target, opt.cppfile, bmifile, {name = name, objectfile = opt.objectfile, requires = opt.module.requires})
 
-                -- needed to detect rebuild of dependencies
-                if provide and build then
-                    mark_build(target, name)
-                end
+            build, dependinfo = should_build(target, opt.cppfile, bmifile, {name = name, objectfile = opt.objectfile, requires = opt.module.requires})
+
+            -- needed to detect rebuild of dependencies
+            if provide and build then
+                mark_build(target, name)
             end
 
             -- append requires flags
             if opt.module.requires then
-                _append_requires_flags(target, opt.module, name, opt.cppfile, bmifile, opt)
+                _append_requires_flags(target, opt.module, name, opt.cppfile, bmifile, table.join(opt, {build = build}))
             end
 
             -- for cpp file we need to check after appendings the flags
-            if build == nil then
+            if not provide and not compiler_support.has_module_extension(opt.cppfile) then
```

Comment:
I'm a little...uncertain about this. It essentially means that, for non-modules, `should_build` gets called *twice*, one before the requires flags are added and one after, In theory, the second one wouldn't be needed, because `_append_requires_flags` seems to always compute the flags from `opt.module.requires`, which was already passed to the first call. However, the above comment explicitly says the check needs to happen *after* the flags are added, which makes me think that there's a distinct reason for that? But I don't know what that reason is, so I just left the second call in place. If I can safely remove this, I'll just update the PR.